validate/validate.go

```diff
@@ -294,10 +294,18 @@ func (v *Validator) CheckMounts() (msgs []string) {
 		return
 	}
 
-	if supportedTypes != nil {
-		for _, mount := range v.spec.Mounts {
-			if !supportedTypes[mount.Type] {
-				msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", mount.Type))
+	for i, fMount := range v.spec.Mounts {
+		if supportedTypes != nil && !supportedTypes[fMount.Type] {
+			msgs = append(msgs, fmt.Sprintf("Unsupported mount type %q", fMount.Type))
+		}
+		for j, bMount := range v.spec.Mounts {
+			if nestedValid(bMount.Destination, fMount.Destination) {
+				if v.spec.Platform.OS == "windows" {
+					msgs = append(msgs, fmt.Sprintf("On Windows, %v nested within %v is forbidden", bMount.Destination, fMount.Destination))
```

Comment:
That's really a problem. The SEPC doesn't say.
I will add limit, let's just check nested mounts first.